tiledb/api/examples/quickstart_sparse_string.rs

```diff
@@ -53,14 +52,16 @@ fn create_array(ctx: &Context) -> TileDBResult<()> {
                 DimensionData {
                     name: "rows".to_owned(),
                     datatype: Datatype::StringAscii,
-                    ..Default::default()
+                    constraints: ().into(),
```

Comment:
Ok, I don't really think `Default` is the right choice actually, but I feel like rquiring the user to write `DimensionConstraint::StringAscii` is not a big ask, and the readability is going to be much better than `().into()`

Reply:
Totes fair on this one as this felt less than super awesome when I was writing it but I couldn't make a decision either direction on what to do. I'll totally rip out that ().into implementation as I agree its even less great looking at it again with fresh eyes, and it wasn't all that great when I wrote it.

I'm not a super big fan of having. to write out `DimensionConstraint::StringAscii` on the line after `Datatype::StringAscii` since it really seems like the constraint should be inferable there. But its at least better than what I've got currently at least.

Reply:
`DimensionConstraint::None` or `DimensionConstraint::Unconstrained`?

I see your point and I kind of agree, especially since the `StringAscii` might well be relaxed in future versions such that other types can also be variable in a dense dimension